mantidproject/mantid
# ILL Reflectometry log overwrite

**Description of work.**

This PR implements an option to replace or create new sample logs to data loaded with `LoadILLReflectometry`. Furthermore, this feature is then connected with reduction workflow algorithms `ReflectometryILLPreprocess` and the high-level `ReflectometryILLAutoProcess`. The implementation assumes that the provided value can be understood as either a number, in which case the value is cast to double, or a string.

Should this feature become interesting to more ILL loaders or beyond, it should be straight-forward to move it to the `LoadHelper` namespace.

**To test:**

1. Make sure that UnitTest data is in your data search directories.
2. Run the following snippet:
```
logs_to_replace = {"ChopperSetting.firstChopper": 2, "ChopperSetting.secondChopper": 1, "new_value": 3.14159}
ws = LoadILLReflectometry('ILL/D17/317369.nxs', LogsToReplace=logs_to_replace)
```

3. Verify that the original values of `ChopperSetting` logs have been successfully replaced with the provided values, and that a new sample log named "new_value" has been created.

Fixes #35279 . 

Documentation and release notes updated.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
